sherpa-onnx/csrc/online-recognizer.cc

```diff
@@ -244,18 +244,40 @@ class OnlineRecognizer::Impl {
     // TODO(fangjun): Remember to change these constants if needed
     int32_t frame_shift_ms = 10;
     int32_t subsampling_factor = 4;
-    return Convert(decoder_result, sym_, frame_shift_ms, subsampling_factor);
+    auto ans = Convert(decoder_result,
+               sym_,
+               frame_shift_ms,
+               subsampling_factor);
+
+    if (!IsReady(s) && s->IsLastFrame(s->NumFramesReady() - 1)) {
+      ans.is_final = true;
+    }
+    ans.segment = s->GetWavSegment();
+    float frame_shift_s = frame_shift_ms / 1000.;
+    ans.start_time = s->GetStartFrame() * frame_shift_s;
+    s->GetNumTrailingBlankFrames() = decoder_result.num_trailing_blanks;
+
+    if (config_.enable_endpoint && IsEndpoint(s)) {
+      auto r = decoder_->GetEmptyResult();
+      s->SetResult(r);
+      ans.is_final = true;
```

Comment:
Why setting `ans.is_final = true` here, I thought this branch is for breaking long audios via `Endpointer`.